programs/ssl/ssl_client2.c

```diff
@@ -343,6 +343,14 @@ int main( void )
 #define USAGE_SERIALIZATION ""
 #endif
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+#define USAGE_EARLY_DATA \
+    "    early_data=%%d        default: 0 (disabled)\n"      \
+    "                        options: 0 (disabled), 1 (enabled)\n"
```

Comment:
```suggestion
    "    early_data=%%d        default: NULL\n"      \
    "                          path of file that contains data.\n"
```

Reply:
Will we pass the early data by this way?  Originally it used set_early_data() to do so.

Reply:
Align with OpenSSL and GnuTLS, similar parameter design.

Reply:
Prefer to leave it to the issue: https://github.com/Mbed-TLS/mbedtls/issues/6340

Reply:
I prefer change it here. Otherwise, we have to pay more loading on `max_early_data_size` test.

Reply:
Can't understand why : "we have to pay more loading on max_early_data_size test." I think there is no related with it currently, at least in this PR.

Reply:
In #6430, it need a parameter for input data, and the parameter can be merged with `early_data` .  If change it here, we do not need add the parameter or change it .  If not, we have to add code in #6430.

And the suggestion is replace `early_data=0` with `early_data=NULL` and `early_data=file path` , that's equivalent  and enough in this PR. I do not think this PR should implement file read. 

That's up to you. If you think it make sense and not need changed in this PR. Please  reference this comment in #6340

Reply:
Hi, the issue should be #6340 .
Sure, I will link this comment in #6340 . Actually, there will be big change in #6340  I am preparing it.

Reply:
This thread is put into #6340,  Please add it and I will close this thread.